### ISSUE TITLE
Support sTestFile1 to sTestFile10 ini properties

### ISSUE THREAD

**Issue by Ortham** [Ortham/libloadorder#84]
As investigated [here](https://github.com/Ortham/libloadorder/issues/81#issuecomment-1730305427), some games allow plugins to be activated by listing them in an ini file instead of in plugins.txt (ignoring Morrowind, which doesn't have plugins.txt). libloadorder doesn't currently support this, so any plugins listed in the ini but not plugins.txt are reported as inactive.

- Oblivion: has two ini files (`Oblivion.ini` and `My Games/Oblivion/Oblivion.ini`) that are mutually exclusive, plugins listed there are active in addition to those in plugins.txt
- Skyrim: has `My Games/Skyrim/Skyrim.ini`, listing anything other than Skyrim.esm and/or Update.esm (even in addition to them) causes the game to crash.
- Skyrim SE: has `My Games/Skyrim Special Edition/Skyrim.ini` with the DLC plugins listed by default, plugins listed there are active in addition to those in plugins.txt and load after the plugins in plugins.txt.
  - The DLC plugins are hardcoded to always be active so listing them in the ini doesn't really add anything.
  - If a plugin is present in plugins.txt and Skyrim.ini, its plugins.txt load order position is used.
  - Plugins are not necessarily loaded in the order that they're listed in Skyrim.ini (even if they're not in plugins.txt).
- Fallout 3: has `My Games/Fallout3/FALLOUT.INI` with Fallout3.esm listed by default, plugins listed there are active in addition to those in plugins.txt
- Fallout New Vegas: has `My Games/FalloutNV/Fallout.ini` with FalloutNV.esm and potentially FalloutNV_lang.esp listed by default, plugins listed there are active in addition to those in plugins.txt
- Fallout 4: has two ini files, `My Games/Fallout4/Fallout4.ini` and `My Games/Fallout4/Fallout4Custom.ini` are merged, with the latter overriding the former if the same property appears in both.
  - Having sTestFile<N> set causes plugins.txt and Fallout4.ccc to be ignored.
  - Plugins are not necessarily loaded in the order that they're listed in the ini files.
- Starfield: has `Starfield.ini`, `Starfield_<lang>.INI` (e.g. `Starfield_en.INI`) and `My Games/Starfield/StarfieldCustom.ini` that are merged similarly to Fallout 4, with `My Games/Starfield/StarfieldCustom.ini` overriding `Starfield_<lang>.ini`, overriding `Starfield.INI`. Plugins.txt is currently not functional and there is currently no Starfield.ccc.

I think it would be best to treat the ini files as read-only, at least to start with, so that any plugin listed in them cannot be deactivated.

I'm not sure what encoding the ini files use, or if it varies between games: libloadorder reads Oblivion.ini as Windows-1252 as that's safe for its current usage, but if it's not accurate then it may misread plugin names, and later games support languages like Japanese that cannot be represented in that encoding. libloadorder currently reads and writes plugins.txt as Windows-1252, and I think I checked that was correct for languages like Japanese, but maybe the ini files are different, as they can contain message strings.

**Comment by Ortham** [comment]
Morrowind: English, French, German

Windows-1252 for all languages.

All languages have different Morrowind.exe files.

---

Oblivion: English, Spanish, French, German, Italian

Windows-1252 for all languages. The game can activate plugins in plugins.txt and Oblivion.ini using Windows-1252.

All languages have the same Oblivion.exe.

---

Skyrim: Spanish, French, German, Italian, Polish, Russian, Czech, Traditional Chinese, Japanese

Polish, Czech, Russian, Japanese and Traditional Chinese have different TESV.exe files from the other languages.

English: Windows-1252 plugins.txt entry works, Windows-1251 and UTF-8 do not
Russian: Windows-1252 plugins.txt entry works, Windows-1251 and UTF-8 do not
Japanese: Windows-1252 plugins.txt entry works, Windows-1251 and UTF-8 do not

---

Skyrim SE: English, Spanish, French, German, Italian, Polish, Russian, Traditional Chinese, Japanese

All languages have the same SkyrimSE.exe.

plugins.txt: Windows-1252 entry works, Windows-1251 and UTF-8 do not
Skyrim.ini: Windows-1252 entry works, Windows-1251 and UTF-8 do not

---

Fallout 3: English, Spanish, French, German, Italian

Windows-1252 for all languages? The default ini only contains English strings for all languages.

All languages have different Fallout3.exe files.

Probably the same as Oblivion, there's no reason to use different encodings.

---

Fallout NV: English, Spanish, French, German, Italian, Polish, Czech, Russian

Windows-1252 for all languages? The default ini only contains English strings for all languages.

All languages have the same FalloutNV.exe.

Probably the same as Skyrim, there's no reason to use different encodings.

---

Fallout 4: English, Spanish, French, German, Italian, Polish, Russian, Brazilian Portuguese, Traditional Chinese, Japanese

Japanese and Traditional Chinese have different Fallout4.exe files from the other languages.

---

Starfield: English, Spanish, French, German, Italian, Polish, Latin American Spanish, Brazilian Portuguese, Traditional Chinese, Japanese

All languages have the same Starfield.exe.

**Comment by Ortham** [comment]
I retested Skyrim's handling of sTestFile<N> and found that it behaves the same way as Skyrim SE - I think the crashes I saw in my previous testing were due to a broken install.

I also retested Fallout 4 but it does work as described above. I did some more testing though and found that having an empty `sTestFile1=` (or any other sTestFile<N>) value or a value that is not a plugin that exists is not enough for the game to ignore plugins.txt and Fallout4.ccc, i.e. it needs to be a real installed plugin.
